lib/_http_outgoing.js

```diff
@@ -602,6 +603,24 @@ OutgoingMessage.prototype.getHeaderNames = function getHeaderNames() {
 };
 
 
+// Returns an array of the names of the current outgoing raw headers.
+OutgoingMessage.prototype.getRawHeaderNames = function getRawHeaderNames() {
+  const headersMap = this[kOutHeaders];
+  const headers = [];
+
+  if (headersMap !== null) {
+    const values = ObjectValues(headersMap);
+    // Retain for(;;) loop for performance reasons
+    // Refs: https://github.com/nodejs/node/pull/30958
+    for (let i = 0, l = values.length; i < l; i++) {
+      headers.push(values[i][0]);
+    }
```

Comment:
```suggestion
    return ArrayPrototypeMap(ObjectValues(headersMap), (value) => value[0]);
```

Reply:
Thanks for the feedback, @aduh95. I'm not entirely sure how using array map vs a for loop could impact performance in this case, but if you take a look at `lib/_http_outgoing.js` all methods handling headers are using this exact code snippet, including the deprecated `_headerNames` one. This is the related issue referred to in the code https://github.com/nodejs/node/pull/30958